cxx-qt-gen/test_outputs/naming.h

```diff
@@ -2,8 +2,15 @@
 
 #include <memory>
 #include <mutex>
+#include <shared_mutex>
+
+namespace rust::cxxqtlib1 {
```

Comment:
I don't think we need this forward declaration if we move the `#include "cxx-qt-gen/include/my_object.cxx.h"` before the `using MyObjectCxxQtThread = rust::cxxqtlib1::CxxQtThread<MyObject>;`

Reply:
I think that failed, as the CXX generated code in the header expects that `MyObjectCxxQtThread` already exists. So i had to do this fun :-)

Reply:
Ah, makes sense, in that case, let's keep it as is.
It's a bit strange how these two files sort off depend on each other :thinking: 

Maybe something to refactor in future as well, so that we generate one file with the forward-declarations needed by `my_object.cxx.h` and then `my_object.cxx.h` can include that.